osu.Game/Skinning/SkinProvidingContainer.cs

```diff
@@ -68,11 +70,35 @@ public SampleChannel GetSample(ISampleInfo sampleInfo)
 
         public IBindable<TValue> GetConfig<TLookup, TValue>(TLookup lookup)
         {
-            if (AllowConfigurationLookup && skin != null)
+            if (skin != null)
             {
-                var bindable = skin.GetConfig<TLookup, TValue>(lookup);
-                if (bindable != null)
-                    return bindable;
+                switch (lookup)
+                {
+                    // todo: the GlobalSkinColours switch is pulled from LegacySkin and should not exist.
+                    // will likely change based on how databased storage of skin configuration goes.
+                    case GlobalSkinColours global:
+                        switch (global)
+                        {
+                            case GlobalSkinColours.ComboColours:
+                                var bindable = skin.GetConfig<TLookup, TValue>(lookup);
+                                if (bindable != null && AllowColourLookup)
+                                    return bindable;
+                                else
+                                    return fallbackSource?.GetConfig<TLookup, TValue>(lookup);
```

Comment:
This pattern is pretty repeatable, the only thing that changes is which flag to check. Would probably make this a local or private method and call that instead of duplicating all over.

Reply:
I agree. I didn't notice the pattern while I was trying to work out the specific calls to make to only pull the colour config items.